### PR TITLE
docs/k8s: updated helm doc for short-lived SA tokens

### PR DESCRIPTION
**Pull request by tsaarni**
Helm docs still have example for Kubernetes Auth Method that store the SA token in configuration https://www.vaultproject.io/docs/platform/k8s/helm/examples/kubernetes-auth. Update the example to allow rotated tokens to be reloaded automatically.

xref #15390

Signed-off-by: Tero Saarni <tero.saarni@est.tech>

### CODE REVIEW

**Comment by tsaarni** [comment]
This is only doc update so skipping the Changelog entry.

I think there is nothing specific to Helm in this doc, so not sure if the page is absolutely necessary, but since the page now exists it would be nice to have up to date info!

File: website/content/docs/platform/k8s/helm/examples/kubernetes-auth.mdx [hunk 1]
```diff
@@ -14,33 +14,18 @@ In this example, we will walk through how to set up the [Kubernetes Auth Method]
...
-
 Exec into the Vault pod:
 
 ```bash
-kubectl exec -it vault-0 /bin/sh
+kubectl exec -it vault-0 -- /bin/sh
 ```
 
-Then run the following command to configure the Kubernetes Auth Method:
+First, login using the `root` token and then run the following commands to configure the Kubernetes Auth Method:
```
Comment (tomhjp) [inline website/content/docs/platform/k8s/helm/examples/kubernetes-auth.mdx:23]:
```suggestion
If you didn't set `server.dev.enabled=true`, you'll need to log in to Vault first using `vault login`. Then run the following commands to configure the Kubernetes Auth Method:
```

**Review by tomhjp — COMMENTED** [review]
Sorry for the delay, this looks like a great update, just one suggestion as I don't want us to encourage non-dev installers to use the root token.

File: website/content/docs/platform/k8s/helm/examples/kubernetes-auth.mdx (hunk 1 above)
Comment (tsaarni) [reply website/content/docs/platform/k8s/helm/examples/kubernetes-auth.mdx]: @tomhjp Thank you for the review. I've fixed according to comments.

**Pull request merged**